source/shared/core_sqlsrv.h

```diff
@@ -2352,4 +2354,26 @@ struct str_conn_attr_func {
     }
 };
 
+struct str_conn_attr_auth_func {
```

Comment:
no need to add a function for authentication, validation and error checking should be done in driver level not core or header, see validate_conn_options function (https://github.com/Microsoft/msphpsql/blob/dev/source/sqlsrv/conn.cpp#L151)